pallets/subtensor/src/swap.rs

```diff
@@ -0,0 +1,422 @@
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `netuid_is_member` - A vector of network IDs where the hotkey is a member.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_loaded_emission(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        netuid_is_member: &[u16],
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        for netuid in netuid_is_member {
+            if let Some(mut emissions) = LoadedEmission::<T>::get(netuid) {
+                for emission in emissions.iter_mut() {
+                    if emission.0 == *old_hotkey {
+                        emission.0 = new_hotkey.clone();
+                    }
+                }
+                LoadedEmission::<T>::insert(netuid, emissions);
+            }
+        }
+        *weight += T::DbWeight::get().writes(netuid_is_member.len() as u64);
+        Ok(())
+    }
+
+    /// Swaps the UIDs of the hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `netuid_is_member` - A vector of network IDs where the hotkey is a member.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_uids(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        netuid_is_member: &[u16],
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        for netuid in netuid_is_member.iter() {
+            if let Ok(uid) = Uids::<T>::try_get(netuid, old_hotkey) {
+                Uids::<T>::remove(netuid, old_hotkey);
+                Uids::<T>::insert(netuid, new_hotkey, uid);
+                weight.saturating_accrue(T::DbWeight::get().writes(2));
+            }
+        }
+        Ok(())
+    }
+
+    /// Swaps the Prometheus data of the hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `netuid_is_member` - A vector of network IDs where the hotkey is a member.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_prometheus(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        netuid_is_member: &[u16],
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        for netuid in netuid_is_member.iter() {
+            if let Ok(prometheus_info) = Prometheus::<T>::try_get(netuid, old_hotkey) {
+                Prometheus::<T>::remove(netuid, old_hotkey);
+                Prometheus::<T>::insert(netuid, new_hotkey, prometheus_info);
+                weight.saturating_accrue(T::DbWeight::get().writes(2));
+            }
+        }
```

Comment:
There's still quite a lot of functions in this file that didn't account for the read in `try_get`.

Reply:
done